pkg/controller/machine.go

```diff
@@ -556,6 +571,9 @@ func parseContainerRuntime(s string) (runtime, version string, err error) {
 }
 
 func (c *Controller) getNode(instance instance.Instance, provider string) (node *corev1.Node, exists bool, err error) {
+	if instance == nil {
```

Comment:
Neither am I, the controller is not supposed to ever run into that case but if it does, it will not not just crash with a NPE.

Reply:
I'm perfectly fine with crashing here with an NPE.
Otherwise we would need to implement such a logic in every function which accepts a pointer. And i highly dislike such kind of protective code. It should be the responsibility of the caller to make sure all parameters are correct.

Or is this any kind of best practice?
I couldn't find similar code in the 
@guusvw @metalmatze

Reply:
Well, errors in an application should only lead to a crash if that error makes the the application defunct unless restarted, which is definitely not the case here.

In this particular case NPEs are somehwat likely to happen because:

* `getNode()` gets called from the `syncHandler` with `providerInstance` as argument
* The very first line of the `syncHandler` is this: `var providerInstance instance.Instance`

=> You can call `getNode(providerInstance)` anywhere inside the `syncHandler` without getting a compiler error but you may cause a NPE

Reply:
I can also call `b, err := ioutil.ReadAll(nil)` anywhere in our code.
`ReadAll` won't check if the parameter is nil. It will exit with an NPE.

I don't expect that `ReadAll` checks my parameter for `nil`. 
If i'm not able to make sure i have correct parameters, i must not call it.

If the `syncHandler` is not able to make sure, that `providerInstance` is not `nil` we must not call `getNode`. Our logic in the `syncHandler` is probably broken then and we must fix it.

Reply:
I agree that fixing this in the `syncHandler` is probably the more appropriate place. But please lets do that in a dedicated PR and lets keep the current quickfix.